compiler_gym/wrappers/core.py

```diff
@@ -124,19 +215,23 @@ class RewardWrapper(CompilerEnvWrapper):
     def reset(self, *args, **kwargs):
         return self.env.reset(*args, **kwargs)
 
-    def step(self, *args, **kwargs):
-        observation, reward, done, info = self.env.step(*args, **kwargs)
-        # Undo the episode_reward update and reapply it once we have transformed
-        # the reward.
-        #
-        # TODO(cummins): Refactor step() so that we don't have to do this
-        # recalculation of episode_reward, as this is prone to errors if, say,
-        # the base reward returns NaN or an invalid type.
-        if reward is not None and self.episode_reward is not None:
-            self.unwrapped.episode_reward -= reward
-            reward = self.reward(reward)
-            self.unwrapped.episode_reward += reward
```

Comment:
I think this workaround needs to be included, because `multistep()` runs the update to `self.episode_reward`.

Reply:
There should really be a unit test to cover this (not saying you need to add it 🙂 , just note to self)

Reply:
I added it back.